### ISSUE TITLE
[Feature Request] Have the builder take what they need next instead of arbitrary resources

### ISSUE THREAD

**Issue by kmarple1** [ldtteam/minecolonies#3183]
Right now it seems like, whenever the builder returns to their hut during a build, they take an arbitrary selection of resources with them, along with the very next thing they need. This is fine for lower level buildings, but for higher level ones, it greatly increases the build time. Given that both the resources needed and the build order are known in advance, it's possible to figure out a semi-optimal order for what should be taken.

Basically, iterate through the blueprint in build order, queuing up the next X stacks of resources used. Leave several slots empty for items removed from terrain and/or existing structures. This should eliminate the builder placing one block and then running back to the hut for what they need next. It won't be optimal unless you can account for blocks that will need to be removed, but it will be far better than the current system.

**Comment by Raycoms** [comment]
Yeah, we got an issue for that already #35 We're going to work on that in the future. Thanks for reporting this

**Comment by kmarple1** [comment]
@Raycoms, they aren't the same issue. #35 is about the deliveryman pre-fetching materials. Mine is about the builder getting the correct materials instead of random ones from their current inventory.
